File: ch3/three_in_one/java/ThreeInOne.java
public class ThreeInOne{
    public static void main(String[] args){
        Stacks<Integer> s = new Stacks<Integer>();
        s.push(1, 0);
        s.push(1, 2);
        s.push(1, 4);
        s.push(2, 55);
        s.push(2, 66);
        s.push(2, 77);
        s.push(3, 1984);
        s.push(1, 5);
        s.push(3, 2000);
        s.p(1);
        s.p(2);
        s.p(3);
        System.out.println("");
        System.out.println(s.top(1));
        s.pop(1);
        System.out.println(s.top(2));
        s.pop(2);
        System.out.println(s.top(3));
        s.pop(3);
        System.out.println("");
        s.p(1);
        s.p(2);
        s.p(3);
    }
}

File: ch3/three_in_one/python/threeInOne.py
class Stacks:
    def __init__(self):
        self.data = []

    def push(self, stack, n):
        self.data.append((stack, n))

    def pop(self, stack):
        for i in range(len(self.data) - 1, -1, -1):
            if(self.data[i][0] == stack):
                self.data.pop(i)
                return

    def top(self, stack):
        for i in range(len(self.data) - 1, -1, -1):
            if(self.data[i][0] == stack):
                return self.data[i]

    def p(self, stack):
        for i in range(len(self.data)):
            if(self.data[i][0] == stack):
                print(self.data[i][1], end=" ")
        print("")

def main():
    s = Stacks()
    s.push(1, 4)
    s.push(1, 5)
    s.push(1, 6)
    s.push(2, 33)
    s.push(2, 44)
    s.push(2, 55)
    s.push(2, 66)
    s.push(1, 7)
    s.push(1, 8)
    s.push(3, 1984)

    s.p(1)
    s.p(2)
    s.p(3)

    print("")

    print(s.top(1))
    s.pop(1)
    print(s.top(2))
    s.pop(2)
    print(s.top(3))
    s.pop(3)

    print("")

    s.p(1)
    s.p(2)
    s.p(3)

if __name__ == "__main__":
    main()
